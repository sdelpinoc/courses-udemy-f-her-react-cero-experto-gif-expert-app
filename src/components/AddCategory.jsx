import { useState } from 'react';
import { PropTypes } from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
    // console.log(setCategories);
    // console.log(categories);

    const [inputValue, setinputValue] = useState('');

    // we receive an event which we destruct in '{ target }'
    const onInputChange = ({ target }) => {
        // console.log(target.value);
        setinputValue(target.value);
    };

    const onSubmit = (event) => {
        // console.log(event);
        event.preventDefault();

        const newInputValue = inputValue.trim();
        // console.log(newInputValue);

        if (newInputValue.length <= 1) return;

        // setCategories(categories => [inputValue, ...categories]);
        onNewCategory(newInputValue);

        setinputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label="form">
            <input
                type="text"
                value={inputValue}
                placeholder="Search gif..."
                // onChange={event => { onInputChange(event) }}
                onChange={onInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
};
