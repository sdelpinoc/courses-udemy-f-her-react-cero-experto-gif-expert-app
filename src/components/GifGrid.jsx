// import { useEffect, useState } from 'react';

// import { getGifs } from '../helpers/getGifs';
import { PropTypes } from 'prop-types';

import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

// Functions must not be executed in a functional component, because every time the component is rendered it is re-executed
// useEffect, hook of react to ejecute secondary effects
export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);
    // console.log({isLoading});

    // getGifs(category);

    // In jsx we use className for the css and not class, because is a reserved word
    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Loading...</h2>)
            }
            <div className="card-grid">
                {
                    images.map(image => (
                        <GifItem
                            key={image.id}
                            {...image} // Spread the properties to the component
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
};