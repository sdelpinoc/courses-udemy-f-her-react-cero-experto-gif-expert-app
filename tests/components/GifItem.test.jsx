import { render, screen } from '@testing-library/react';

import { GifItem } from '../../src/components/GifItem';

describe('Testing in <GifItem />', () => {

    const title = 'One Punch';
    const url = 'https://one-punch.com/saitama.jpg';

    test('Should made match with the snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);

        expect(container).toMatchSnapshot();
    });

    test('Should check the image source(src) and the ALT', () => {
        render(<GifItem title={title} url={url} />);
        // screen.debug();

        // expect(screen.getByRole('img').src).toBe(url);
        // expect(getByRole('img', { name: title }).src).toContain(url);
        const { src, alt } = screen.getByRole('img');
        // console.log(src);
        // console.log(alt);

        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Should show the title in the component', () => {
        render(<GifItem title={title} url={url} />);

        // Only search in texts, not attributes or properties like 'alt'
        expect(screen.getByText(title)).toBeTruthy();
    });
});