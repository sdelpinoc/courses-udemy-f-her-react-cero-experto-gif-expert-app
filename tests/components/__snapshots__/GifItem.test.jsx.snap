// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <GifItem /> Should made match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch"
      src="https://one-punch.com/saitama.jpg"
    />
    <p>
      One Punch
    </p>
  </div>
</div>
`;
